Khan/perseus
# [Interactive Graph Editor] Add UI for updating starting coordinates (linear and ray)

## Summary:
Implement a UI in the interactive graph editor to allow conten authors
to set custom starting coordinates for interactive graphs. This is a
prerequisite for hint mode.

Note: There was some funky stuff happening when I tried using `coords`.
I added a `startCoords` field to the graph types, and that seemed to fix
all my issues. This change is included in this PR.

Included in this PR
- The creation of the StartCoordSettings component
- The flag that this feature is behind
- The UI itself, only for linear and ray graphs
- Added `startCoords` field

Issue: https://khanacademy.atlassian.net/browse/LEMS-2052

## Test plan:
Storybook existing starting coords
- http://localhost:6006/?path=/story/perseuseditor-editorpage--interactive-graph-linear-with-starting-coords
- Confirm that the UI reflects the custom coords
- Change the coords
- Confirm that the graph changes
- Change the graph type to Ray
- Confirm that the coords change to the ray initial state coords
- Confirm that the "correct" preview is still interact-able and does not interfere with the start coords preview 
- Press "Use default start coords" and confirm it uses the start coords

Storybook new graph
- http://localhost:6006/?path=/story/perseuseditor-editorpage--demo
- Add a new interactive graph
- Select the linear graph type
- Confirm that it sets the default coords in the UI
- Confirm that the "correct" preview is still interact-able and does not interfere with the start coords preview

Copy/pasting
- Press the "+ Add a hint" button at the bottom of the page
- Copy/paste the interactive graph ("[[interactive-graph 1]]")
  into the hint text box
- Confirm that the graph is copied correctly with the same
  start coords
- Confirm that the "correct" preview is still interact-able and does not interfere with the start coords preview
- Press "Use default start coords" and confirm it uses the start coords


![Screenshot 2024-06-28 at 5 18 33 PM](https://github.com/Khan/perseus/assets/13231763/90ba9c4c-6029-4684-8701-90d38b655a15)
